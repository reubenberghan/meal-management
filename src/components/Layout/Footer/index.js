import * as React from 'react'

import {
  COOKIE_POLICY_PATH,
  PRIVACY_POLICY_PATH,
  TERMS_OF_USE_PATH
} from '../../../state/constants'
import makeLocalLink from '../../../wrappers/makeLocalLink'

import Copyright from './Copyright'
import FooterLink from './FooterLink'
import FooterNav from './FooterNav'
import FooterWrapper from './FooterWrapper'

const Link = makeLocalLink(FooterLink)

export default function Footer () {
  return (
    <FooterWrapper>
      <Copyright />
      <FooterNav>
        <Link href={COOKIE_POLICY_PATH}>Cookie Policy</Link>
        <Link href={PRIVACY_POLICY_PATH}>Privacy Policy</Link>
        <Link href={TERMS_OF_USE_PATH}>Terms of Use</Link>
      </FooterNav>
    </FooterWrapper>
  )
}
