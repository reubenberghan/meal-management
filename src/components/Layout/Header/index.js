import * as React from 'react'

import {
  ABOUT_PATH,
  HOME_PATH,
  RECIPES_PATH,
  SIGN_IN_PATH
} from '../../../state/constants'
import makeLocalLink from '../../../wrappers/makeLocalLink'

import HeaderLink from './HeaderLink'
import HeaderNav from './HeaderNav'
import HeaderWrapper from './HeaderWrapper'
import Logo from './Logo'

const Link = makeLocalLink(HeaderLink)

export default function Header () {
  return (
    <HeaderWrapper>
      <Logo />
      <HeaderNav>
        <Link href={HOME_PATH}>Home</Link>
        <Link href={ABOUT_PATH}>About</Link>
        <Link href={RECIPES_PATH}>Recipes</Link>
        <Link href={SIGN_IN_PATH}>Sign In</Link>
      </HeaderNav>
    </HeaderWrapper>
  )
}
