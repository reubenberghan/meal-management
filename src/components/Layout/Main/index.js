import styled from 'styled-components'

const Main = styled.main`
  margin: 5rem 1rem;
  grid-area: main;
`
Main.displayName = 'Main'

export default Main
