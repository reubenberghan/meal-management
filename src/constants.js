export const SITE_NAME = 'Meal Management'
