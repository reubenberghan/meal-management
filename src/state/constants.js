// Actions
export const RECIPE_ADDED = '@@recipes/RECIPE_ADDED'
export const TITLE_CHANGED = '@@addRecipe/TITLE_CHANGED'
export const INGREDIENT_CHANGED = '@@addRecipe/INGREDIENT_CHANGED'
export const STEP_CHANGED = '@@addRecipe/STEP_CHANGED'

// Paths
export const ABOUT_PATH = '/about'
export const COOKIE_POLICY_PATH = '/cookie-policy'
export const HOME_PATH = '/'
export const PRIVACY_POLICY_PATH = '/privacy-policy'
export const RECIPES_PATH = '/recipes'
export const SIGN_IN_PATH = '/sign-in'
export const TERMS_OF_USE_PATH = '/terms-of-use'

// State
export const initialState = {
  recipes: [
    {
      id: '123',
      title: 'Banana Cake',
      ingredients: [
        {
          id: 'abc',
          name: 'Banana'
        },
        {
          id: 'abd',
          name: 'Flour'
        }
      ],
      steps: [
        {
          id: 'abe',
          step: 'Peel banana'
        },
        {
          id: 'abf',
          step: 'Mix ingredients'
        }
      ]
    },
    {
      id: '124',
      title: 'Toasted Sandwich',
      ingredients: [
        {
          id: 'abg',
          name: 'Bread'
        },
        {
          id: 'abh',
          name: 'Cheese'
        }
      ],
      steps: [
        {
          id: 'abi',
          step: 'Slice cheese'
        },
        {
          id: 'abj',
          step: 'Toast bread'
        }
      ]
    }
  ]
}
