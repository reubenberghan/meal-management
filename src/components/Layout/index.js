import * as React from 'react'
import { Fragment } from 'react'
import Helmet from 'react-helmet'

import { SITE_NAME } from '../../constants'

export default function Layout ({ children, title }) {
  const pageTitle = title ? `${title} - ${SITE_NAME}` : SITE_NAME

  return (
    <Fragment>
      <Helmet>
        <title>{pageTitle}</title>
      </Helmet>
      {children}
    </Fragment>
  )
}
