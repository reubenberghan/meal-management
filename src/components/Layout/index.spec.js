import * as React from 'react'
import { shallow } from 'enzyme'

import { SITE_NAME } from '../../constants'

import Layout from '.'

describe('components:Layout', () => {
  it(`should render the Layout with ${SITE_NAME} when no title prop`, () => {
    expect(toJson(shallow(<Layout />))).toMatchSnapshot()
  })

  it(`should render the Layout ${SITE_NAME} prefixed by title prop when given`, () => {
    expect(toJson(shallow(<Layout title='About' />))).toMatchSnapshot()
  })
})
