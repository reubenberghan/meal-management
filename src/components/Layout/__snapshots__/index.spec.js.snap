// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Layout should render the Layout Meal Management prefixed by title prop when given 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      About - Meal Management
    </title>
  </HelmetWrapper>
</Fragment>
`;

exports[`components:Layout should render the Layout with Meal Management when no title prop 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Meal Management
    </title>
  </HelmetWrapper>
</Fragment>
`;
